spring/narayana-spring-boot/pom.xml

```diff
@@ -73,6 +77,11 @@
     </dependencyManagement>
 
     <dependencies>
+        <dependency>
+            <groupId>org.javassist</groupId>
+            <artifactId>javassist</artifactId>
+            <version>3.25.0-GA</version>
+        </dependency>
```

Comment:
Just out of curiosity, why this dependency?

Reply:
This is an interesting question @jmfinelli! This quickstart uses hibernate-core dependency (from spring-boot-starter-data-jpa). hibernate-core contains javassist version 3.22.0-GA that is not working fine with JDK17, so I added version 3.25.0-GA that works.
I would prefer to update the springframework dependencies ( here spring-boot-dependencies is at version 2.0.9.RELEASE) to a higher version but in latest version some classes used in this quickstart are missing (i.e. org.springframework.boot.jta.narayana.NarayanaRecoveryManagerBean). @mmusgrov Do you think these classes (i.e. NarayanaRecoveryManagerBean, DataSourceXAResourceRecoveryHelper, ...) should be included in next spring-boot releases?

Reply:
I will add a comment in the pom to clarify this.

Reply:
What about if you exclude `javassist` instead of adding it?

Reply:
Unfortunately spring uses that dependency. Removing it causes a NoClassDefFoundError:
`org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'entityManagerFactory' defined in class path resource [org/springframework/boot/autoconfigure/orm/jpa/HibernateJpaConfiguration.class]: Invocation of init method failed; nested exception is java.lang.NoClassDefFoundError: javassist/bytecode/ClassFile`

Reply:
Thanks for double checking this +1